File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.js
var sumNumbers = function(root) {
    let ans=0;
    let sum =0;
    let solve =(root, ans) => {
    if(!root)return 0;
    ans=ans*10 + root.val;
    if(!root.left&&!root.right)sum+=ans;
    let left=solve(root.left,ans);
    let right=solve(root.right,ans);

    return ans;
}
    solve(root,ans);
    return sum;
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
class Solution {
public:
    int sum =0;
    int solve(TreeNode* root,int ans){
        if(!root)return 0;
        ans=ans*10 + root->val;
        if(!root->left&&!root->right)sum+=ans;
        int left=solve(root->left,ans);
        int right=solve(root->right,ans);
        
        return ans;
    }
    int sumNumbers(TreeNode* root) {
        int ans=0;
        solve(root,ans);
        return sum;
    }
};